src/pages/user/InstitutionSelector.tsx

```diff
@@ -10,89 +10,88 @@ import InstitutionSearch from '../publication/references_tab/components/Institut
-import InstitutionPresentation from './InstitutionPresentation';
 
 const StyledInstitutionSelector = styled.div`
   width: 30rem;
 `;
 
 interface InstitutionSelectorProps {
   setSelectedCristinUnitId: (value: string) => void;
+  disabled: boolean;
 }
 
-const InstitutionSelector: React.FC<InstitutionSelectorProps> = ({ setSelectedCristinUnitId }) => {
+const InstitutionSelector: React.FC<InstitutionSelectorProps> = ({ setSelectedCristinUnitId, disabled }) => {
   const { t } = useTranslation('profile');
-  const [faculties, setFaculties] = useState<InstitutionSubUnit[]>([]);
-  const [institutes, setInstitutes] = useState<InstitutionSubUnit[]>([]);
+  const [subUnits, setSubUnits] = useState<InstitutionSubUnit[]>([]);
   const [selectedInstitution, setSelectedInstituion] = useState(emptyInstitution);
-  const [selectedFaculty, setSelectedFaculty] = useState(emptyInstitutionSubUnit);
-  const [selectedInstitute, setSelectedInstitute] = useState(emptyInstitutionSubUnit);
+  const [subUnitSelector, setSubUnitSelector] = useState<InstitutionSubUnit>();
 
-  const getSubUnits = async (searchValue: string, setStateFunction: (value: InstitutionSubUnit[]) => void) => {
-    const response = await getInstitutionSubUnit(searchValue);
+  const getSubUnit = async (cristinUnitId: string) => {
+    const newSubUnit: InstitutionSubUnit = emptyInstitutionSubUnit;
+
+    const response: InstitutionSubUnit[] = await getInstitutionSubUnit(cristinUnitId);
     if (response) {
-      setStateFunction(response);
+      newSubUnit.subUnits = response;
     }
-  };
 
-  const resetInstitute = () => {
-    setInstitutes([]);
-    setSelectedInstitute(emptyInstitutionSubUnit);
+    return newSubUnit;
   };
 
-  const resetFaculty = () => {
-    setFaculties([]);
-    setSelectedFaculty(emptyInstitutionSubUnit);
-    resetInstitute();
-  };
-
-  const searchFaculties = (institution: Institution) => {
+  const selectInstitution = async (institution: Institution) => {
     setSelectedCristinUnitId(institution.cristinUnitId);
     setSelectedInstituion(institution ?? selectedInstitution);
-    resetFaculty();
-    getSubUnits(institution.cristinUnitId, setFaculties);
+    setSubUnits([]);
+    const searchValue = await getSubUnit(institution.cristinUnitId);
+    setSubUnitSelector(searchValue);
   };
 
-  const searchInstitutes = (faculty: InstitutionSubUnit) => {
-    setSelectedCristinUnitId(faculty.cristinUnitId);
-    setSelectedFaculty(faculty);
-    resetInstitute();
-    getSubUnits(faculty.cristinUnitId, setInstitutes);
-  };
+  const searchSubUnits = async (newSubUnit: InstitutionSubUnit, index: number) => {
+    setSelectedCristinUnitId(newSubUnit.cristinUnitId);
+    let oldSearchValues = subUnitSelector?.subUnits || [];
+    if (index < subUnits.length) {
+      oldSearchValues = subUnits[index].subUnits || [];
+    }
+    const searchValue = await getSubUnit(newSubUnit.cristinUnitId);
+    newSubUnit.subUnits = [...oldSearchValues];
```

Comment:
`oldSearchValues` er allerede array, så trenger ikke å spreade ut for så å pakke inn igjen. Samme for `subUnits` noen rett under